WordPress/gutenberg
# Editor: Extract BlockList as reusable component

This pull request seeks to extract block list components from `VisualEditor`, for purpose of reusability, particularly for managing nested block listing.

__Testing instructions:__

Verify that there are no remaining lingering references to visual editor for what had previously targeted block list components (unfortunately there are many implicit dependencies, which should be minimized in future refactoring).

Verify that there is no regressions in block listing behavior.
